generator/feature_sorter.cpp

```diff
@@ -155,17 +154,23 @@ class FeaturesCollector2 : public FeaturesCollector
     bool const isArea = fb.IsArea();
     CHECK(!isLine || !isArea, (fb.GetMostGenericOsmId()));
 
+    Points points;  // take out from the loop below to keep allocated vector's memory
+    m2::RectD const & rect = fb.GetLimitRect();
+    Polygons const & polys = fb.GetGeometry();
+
     int const scalesStart = static_cast<int>(m_header.GetScalesCount()) - 1;
     for (int i = scalesStart; i >= 0; --i)
     {
       int level = m_header.GetScale(i);
-      // TODO : re-checks geom limit rect size via IsDrawableForIndexGeometryOnly() which was checked already in CalculateMidPoints.
+      /// @todo: Re-checks geom limit rect size via IsDrawableForIndexGeometryOnly()
+      /// which was already checked in CalculateMidPoints.
       if (fb.IsDrawableInRange(scales::PatchMinDrawableScale(i > 0 ? m_header.GetScale(i - 1) + 1 : 0),
                                scales::PatchMaxDrawableScale(level)))
       {
         bool const isCoast = fb.IsCoastCell();
```

Comment:
take out of the loop too?

Reply:
No, only for coastlines. I think we can (try to) remove this check since we increased the coast's precision.